### PR TITLE
Refactor fritz utility to use python-fire

### PR DESCRIPTION
**Pull request by dmitryduev**
In this PR:

- Refactored the `fritz` utility to use `python-fire` instead of `argparse`. This does not touch the logic in `fritz` yet, it is meant to improve the code's readability and maintainability, e.g. we will now be able to utilize typing.
- The `fritz` utility now first checks if python requirements are satisfied in addition to checking system dependencies.

### CODE REVIEW

**Comment by dmitryduev** [comment]
Building docs on CI fails because of [this requirement](https://github.com/cesium-ml/baselayer/blob/9913cb6a3595c596ea54a21071d7a312f490789b/requirements.txt#L6) in BL.

**Comment by dmitryduev** [comment]
@dannygoldstein @acrellin ready for review when you have a moment.

**Comment by dmitryduev** [comment]
@dannygoldstein: doc build currently fails on CI bc the pinned BL version lacks the `pyjwt` fix that went in a couple days ago.

**Comment by acrellin** [comment]
this is failing for me in a new way locally when running `./fritz run --init`: 
```
[✗] Checking configuration                                                                                
Traceback (most recent call last):                                                                        
  File "kowalski.py", line 532, in <module>                                                               
    fire.Fire(Kowalski)                                                                                   
  File "/home/arien/anaconda3/envs/fritz/lib/python3.8/site-packages/fire/core.py", line 141, in Fire     
    component_trace = _Fire(component, args, parsed_flag_args, context, name)                     [0/1854]
  File "/home/arien/anaconda3/envs/fritz/lib/python3.8/site-packages/fire/core.py", line 466, in _Fire    
    component, remaining_args = _CallAndUpdateTrace(                                                      
  File "/home/arien/anaconda3/envs/fritz/lib/python3.8/site-packages/fire/core.py", line 681, in _CallAndU
pdateTrace                                                                                                
    component = fn(*varargs, **kwargs)                                                                    
  File "kowalski.py", line 292, in build                                                                  
    check_configs(config_wildcards=config_wildcards)                                                      
  File "kowalski.py", line 157, in check_configs                                                          
    raise KeyError("Fix config.yaml before proceeding")                                                   
KeyError: 'Fix config.yaml before proceeding'                                                             
Traceback (most recent call last):                                                                        
  File "./fritz", line 861, in <module>                                                                   
    fire.Fire(Fritz)                                                                                      
  File "/home/arien/anaconda3/envs/fritz/lib/python3.8/site-packages/fire/core.py", line 141, in Fire     
    component_trace = _Fire(component, args, parsed_flag_args, context, name)                             
  File "/home/arien/anaconda3/envs/fritz/lib/python3.8/site-packages/fire/core.py", line 466, in _Fire    
    component, remaining_args = _CallAndUpdateTrace(                                                      
  File "/home/arien/anaconda3/envs/fritz/lib/python3.8/site-packages/fire/core.py", line 681, in _CallAndU
pdateTrace                                                                                                
    component = fn(*varargs, **kwargs)                                                                    
  File "./fritz", line 579, in run                                                                        
    self.build(                                                                                           
  File "./fritz", line 386, in build                                                                      
    raise RuntimeError("Failed to build Kowalski's docker images")                                        
RuntimeError: Failed to build Kowalski's docker images                                                    

```

**Comment by dmitryduev** [comment]
@acrellin you should prob rm (or better mv) your `fritz.yaml` before trying this.

**Comment by acrellin** [comment]
after running `fritz prune`, which de-inits submodules, `fritz run --init` fails: 
```
(fritz) ➜  fritz git:(fc4996f) ✗ ./fritz run --init                                                       
Checking system dependencies:                                                                             
[✓] python >= 3.7                 [   3.8.8]                                                              
[✓] docker >= 18.06               [19.03.13]                                                              
[✓] docker-compose >= 1.22.0      [  1.25.0]                                                              
Checking python requirements:                                                                             
Traceback (most recent call last):                                                                        
  File "./fritz", line 855, in <module>                                                                   
    env_ok = dependencies_ok()                                                                            
  File "/home/arien/projects/fritz/tools/check_environment.py", line 114, in dependencies_ok              
    for requirement in get_python_requirements():                                                         
  File "/home/arien/projects/fritz/tools/check_environment.py", line 55, in get_python_requirements       
    get_python_requirements(                                                                              
  File "/home/arien/projects/fritz/tools/check_environment.py", line 51, in get_python_requirements       
    with open(requirements_path / requirements_file_name) as requirements_file:                           
FileNotFoundError: [Errno 2] No such file or directory: '/home/arien/projects/fritz/kowalski/requirements.
txt'                                                                                                      
```

**Review by acrellin — APPROVED** [review]
Looks good, aside from above comment regarding the inability to run the utility w/out having first initialized submodules (which I believe is something the utility itself is supposed to handle)

**Comment by dmitryduev** [comment]
@acrellin, ah, nice catch, will fix.

**Comment by dmitryduev** [comment]
@acrellin should be fixed now, could you please check?

File: fritz [hunk 1]
```diff
@@ -9,310 +8,596 @@ import requests
...
+            "JWT_ALGORITHM": jwt_algorithm,
+            "JWT_EXP_DELTA_SECONDS": jwt_exp_delta_seconds,
+        }
+
+        payload = {
+            "user_id": jwt_config["user_id"],
+            "exp": (
+                datetime.datetime.utcnow()
+                + datetime.timedelta(seconds=jwt_config["JWT_EXP_DELTA_SECONDS"])
+            ),
```
Comment (kmshin1397) [inline fritz:211]: Do we need the expiration date here? Maybe pin the latest K and get rid of it now that it's optional?

File: fritz [hunk 2]
```diff
@@ -9,310 +8,596 @@ import requests
...
+    if do_initialize:
+        p = subprocess.run(["git", "submodule", "update", "--init", "--recursive"])
+        if p.returncode != 0:
+            raise RuntimeError("Failed to initialize fritz's submodules")
+
 
-    return jwt_token
+class Fritz:
+    # def __init__(self):
+    #     pass
```
Comment (kmshin1397) [inline fritz:33]: Can delete commented out lines?

**Review by kmshin1397 — APPROVED** [review]
Looks good! Just a couple of minor comments

File: fritz (hunk 1 above)
Comment (dmitryduev) [reply fritz]: Ah, thanks for catching this, made it optional.

File: fritz (hunk 2 above)
Comment (dmitryduev) [reply fritz]: Oops, removed.

**Comment by dmitryduev** [comment]
@kmshin1397 @acrellin the CI is finally green (and it successfully builds/spins up from scratch + tests pass locally) -- please merge if you're happy with it.

**Comment by dmitryduev** [comment]
Actually, @stefanv, would you mind taking a quick look?

**Comment by stefanv** [comment]
@dmitryduev All the `@staticmethod`s are pretty ugly; we could move those out of the class.  Also, I wonder if we shouldn't consider using `fire.Fire()` or, even better, `fire.Fire({dictionary: specification})` instead of plopping all our functions into a class.  The extra indentation doesn't help, and there's no good reason these should all be in the same class (as evidenced by all the static methods).

**Comment by dmitryduev** [comment]
Yea, I agree @stefanv, will fix.

**Comment by dmitryduev** [comment]
@stefanv done. Better? :)

File: fritz [hunk 3]
```diff
@@ -128,145 +185,85 @@ def check_config(cfg="fritz.defaults.yaml", yes=False, **kwargs):
...
+    branch: str = "master",
+    traefik: bool = False,
+    no_kowalski: bool = False,
+    do_update: bool = False,
+    skyportal_tag: str = "skyportal/web:latest",
+    yes: bool = False,
+):
+    """Build Fritz
+
+    :param init: Initialize before updating Fritz
```
Comment (stefanv) [inline fritz:245]: This seems different from everywhere else

**Review by stefanv — APPROVED** [review]
Thanks, Dima, let's merge and then we can keep refactoring from there.

File: fritz (hunk 3 above)
Comment (dmitryduev) [reply fritz]: Fixed the description.

**Comment by dmitryduev** [comment]
Self-merging with 3 blessings.

**Pull request merged**